ruby_oop/mastermind/game.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require './computer_player'
+require './human_player'
+
+# class to play the mastermind game
+class Game
+  attr_accessor :human_player, :computer_player
+
+  AVAILABLE_COLORS = ['🟤', '🔴', '🔵', '🟢', '🟡', '🟣'].freeze
+  @secret_code = []
+
+  def initialize
+    @human_player = codebreaker_or_codemaker
+    @computer_player = !@human_player
+  end
+
+  def computer_codemaker
+    puts "Computer is selecting random colors...\n\n"
+    @secret_code = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@secret_code)
+  end
+
+  def computer_codebreaker
+    puts "Computer is selecting random colors...\n\n"
+    @guess_array = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@guess_array)
+  end
```

Comment:
These are basically duplicative, I don't love it but idk how to DRY it up, open to feedback/pairing if someone has thoughts.

Reply:
This might be a good opportunity to dive into your thread regarding `public` and `private` methods in more detail.  I'd be happy to pair on this!